pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -0,0 +1,743 @@
+					unavailableClaims = append(unavailableClaims, index)
+				}
+			}
+		case claim.Status.DeallocationRequested:
+			// We shouldn't get here. PreFilter already checked this.
+			return statusUnschedulable(logger, "resourceclaim must be reallocated", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclaim", klog.KObj(claim))
+		default:
+			// This must be delayed allocation. Immediate
+			// allocation was already checked for in PreFilter.
+
+			// The ResourceClass might have a node filter. This is
+			// useful for trimming the initial set of potential
+			// nodes before we ask the driver(s) for information
+			// about the specific pod.
+			class, err := pl.classLister.Get(claim.Spec.ResourceClassName)
+			if err != nil {
+				// If the class does not exist, then allocation cannot proceed.
+				return statusError(logger, fmt.Errorf("look up resource class: %v", err))
+			}
+			if class.SuitableNodes != nil {
+				matches, err := corev1helpers.MatchNodeSelectorTerms(node, class.SuitableNodes)
+				if err != nil {
+					return statusError(logger, fmt.Errorf("potential node filter: %v", err))
+				}
+				if !matches {
+					return statusUnschedulable(logger, "excluded via potential node filter in resource class", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclass", klog.KObj(class))
+				}
+			}
+
+			// Now we need information from drivers.
+			podScheduling, err := state.initializePodScheduling(ctx, pod, pl.podSchedulingLister)
+			if err != nil {
+				return statusError(logger, err)
+			}
+			status := statusForClaim(podScheduling, pod.Spec.ResourceClaims[index].Name)
+			if status != nil {
+				for _, unsuitableNode := range status.UnsuitableNodes {
+					if node.Name == unsuitableNode {
+						return statusUnschedulable(logger, "resourceclaim cannot be allocated for the node (unsuitable)", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclaim", klog.KObj(claim), "unsuitablenodes", status.UnsuitableNodes)
+					}
+				}
+			}
+		}
+	}
+
+	if len(unavailableClaims) > 0 {
+		state.mutex.Lock()
+		defer state.mutex.Unlock()
+		if state.unavailableClaims == nil {
+			state.unavailableClaims = sets.NewInt()
+		}
+		state.unavailableClaims.Insert(unavailableClaims...)
+		return statusUnschedulable(logger, "resourceclaim not available on the node", "pod", klog.KObj(pod))
+	}
+
+	return nil
+}
+
+// PostFilter checks whether freeing an allocated claim might help to get a Pod
+// schedulable. This only gets called when filtering found no suitable node.
+func (pl *dynamicResources) PostFilter(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, filteredNodeStatusMap framework.NodeToStatusMap) (*framework.PostFilterResult, *framework.Status) {
+	logger := klog.FromContext(ctx)
+	state, err := getStateData(cs)
+	if err != nil {
+		return nil, statusError(logger, err)
+	}
+	if len(state.claims) == 0 {
+		return nil, nil
+	}
```

Comment:
And I suppose the reason is irrelevant because its not getting shown? Added `framework.NewStatus(framework.UnschedulableAndUnresolvable, "nothing done, no claims")` and `framework.NewStatus(framework.UnschedulableAndUnresolvable, "still not schedulable")`